scratch_kotlin_lib/src/main/java/dev/sgrem/scratch_kotlin_lib/Week3Homework.kt

```diff
@@ -0,0 +1,243 @@
+    println("Grade for score $score: ${getGrade(score)}")
+    score = 79
+    println("Grade for score $score: ${getGrade(score)}")
+    score = 69
+    println("Grade for score $score: ${getGrade(score)}")
+    score = 59
+    println("Grade for score $score: ${getGrade(score)}")
+    score = 101
+    println("Grade for score $score: ${getGrade(score)}")
+    println()
+
+    println("6. Nullability")
+    var myInt: Int? = 5
+    println("${myInt ?: 0} * 2 = ${timesTwo(myInt)}")
+    myInt = null
+    println("${myInt ?: 0} * 2 = ${timesTwo(myInt)}")
+    println()
+
+    println("7. (Above and Beyond) Spiral Array:")
+    var n = 1
+    println("Spiral $n x $n matrix:")
+    printListOfLists(generateSpiralArray(n))
+    n = 2
+    println("Spiral $n x $n matrix:")
+    printListOfLists(generateSpiralArray(n))
+    n = 3
+    println("Spiral $n x $n matrix:")
+    printListOfLists(generateSpiralArray(n))
+    n = 4
+    println("\nSpiral $n x $n matrix:")
+    printListOfLists(generateSpiralArray(n))
+    n = 5
+    println("\nSpiral $n x $n matrix:")
+    printListOfLists(generateSpiralArray(n))
+
+}
+
+
+/*
+1. Boolean Comparisons:
+- Write a Kotlin function that takes two integer parameters and returns `true` if their sum is divisible by 4, otherwise returns `false`.
+*/
+fun isDivisibleByFour(num1: Int, num2: Int): Boolean {
+    return (num1 + num2) % 4 == 0
+}
+
+
+/*
+2. Combining Logic Operators:
+- Write a Kotlin function that takes in a string parameter and returns `true` if the string contains both ‘a’ and ‘z’ characters (the check is case-sensitive), otherwise returns `false`.
+*/
+fun String.containsBoth(a: Char, b: Char): Boolean {
+    return this.contains(a) && this.contains(b)
+}
+
+
+/*
+3. If/Else Conditions:
+- Write a Kotlin function called checkTemperature that takes a temperature in Celsius as a parameter. The function should return a message based on the temperature as follows:
+If the temperature is below freezing (0°C or lower), return "It's freezing cold!"
+If the temperature is between 0°C and 10°C (inclusive), return "It's chilly."
+If the temperature is between 11°C and 25°C (inclusive), return "It's pleasant."
+If the temperature is above 25°C, return "It's warm."
+*/
+fun checkTemperature(temperatureCelsius: Int): String {
+    return when {
+        temperatureCelsius <= 0 -> "It's freezing cold!"
+        temperatureCelsius in 1..10 -> "It's chilly."
+        temperatureCelsius in 11..25 -> "It's pleasant."
+        else -> "It's warm."
+    }
+}
+
+
+/*
+4. Loops:
+- Write a Kotlin function called printMultiples that takes two integers, number and limit, as parameters. The function should use a loop to print all the multiples of number up to and including limit. Each multiple should be printed on a new line.
+In this example, when printMultiples(3, 15) is called, the function should print:
+3
+6
+9
+12
+15
+*/
+fun printMultiples(number: Int, limit: Int) {
+    for (i in number..limit step number) {
```

Comment:
You can also use "multiple" instead of "i" to be more explicit. We try to avoid naming loop variables as i, j, k.